vm/ubpf_jit_arm64.c

```diff
@@ -1005,17 +991,15 @@ translate(struct ubpf_vm *vm, struct jit_state *state, char **errmsg)
 }
 
 static void
-divmod(struct jit_state *state, uint16_t pc, uint8_t opcode, int rd, int rn, int rm)
+divmod(struct jit_state *state, uint8_t opcode, int rd, int rn, int rm)
 {
     bool mod = (opcode & EBPF_ALU_OP_MASK) == (EBPF_OP_MOD_IMM & EBPF_ALU_OP_MASK);
     bool sixty_four = (opcode & EBPF_CLS_MASK) == EBPF_CLS_ALU64;
     enum Registers div_dest = mod ? temp_div_register : rd;
 
-    // Handle divide by zero case
-    // CBNZ rm, .+12 (we don't need to note_jump() as we know the destination immediately).
-    emit_instruction(state, (sixty_four << 31) | CBR_CBNZ | (3 << 5) | rm);
-    emit_movewide_immediate(state, true, caller_saved_registers[2], pc);
-    emit_unconditionalbranch_immediate(state, UBR_B, TARGET_PC_DIV_BY_ZERO);
+    /* Do not need to treet divide by zero as special because the UDIV instruction already
+     * returns 0 when dividing by zero.
```

Comment:
What about mod? What is the value returned by:
X mod 0

BPF spec requires:
X mod 0 == X

Reply:
Running tests now as https://github.com/iovisor/ubpf/pull/124. If they pass, then the ARM64 and AMD64 behavior match.

Reply:
Great, looks like the arm64 div/mod behavior matches BPF behavior.